src/topp/MapAlignerTreeGuided.cpp

```diff
@@ -0,0 +1,581 @@
+    buildTree_(feature_maps, tree, maps_ranges);
+
+    // print tree
+    ClusterAnalyzer ca;
+    OPENMS_LOG_INFO << "  Alignment follows tree: " << ca.newickTree(tree, true) << endl;
+
+    vector<Size> trafo_order;
+    FeatureMap map_transformed;
+    // alignment uses copy of feature_maps, returning result within map_transformed (to keep original data)
+    treeGuidedAlignment_(tree, feature_maps, maps_ranges, map_transformed, trafo_order, model_params, model_type);
+
+    //-------------------------------------------------------------
+    // generating output
+    //-------------------------------------------------------------
+    vector<TransformationDescription> transformations(in_files_size); // for trafo_out
+    computeTransformationsByOrigInFeature_(feature_maps, map_transformed, transformations, trafo_order, model_params, model_type);
+    computeTransformedFeatureMaps_(feature_maps, transformations);
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+    // store transformed feature_maps
+    storeFeatureXMLs_(feature_maps, out_files, fxml_file);
+
+    // store transformations
+    storeTransformationDescriptions_(transformations, out_trafos);
+
+    return EXECUTION_OK;
+  }
+};
+
+template<typename MapType>
+void TOPPMapAlignerTreeGuided::loadInputMaps_(vector<MapType>& maps, StringList& ins, FeatureXMLFile& fxml_file)
+{
+  ProgressLogger progresslogger;
+  progresslogger.setLogType(TOPPMapAlignerBase::log_type_);
+  progresslogger.startProgress(0, ins.size(), "loading input files");
+  for (Size i = 0; i < ins.size(); ++i)
+  {
+    progresslogger.setProgress(i);
+    fxml_file.load(ins[i], maps[i]);
+  }
+  progresslogger.endProgress();
+}
+
+void TOPPMapAlignerTreeGuided::getPeptideSequences_(const vector<PeptideIdentification>& peptides,
+        TOPPMapAlignerTreeGuided::SeqAndRTList& peptide_rts, vector<double>& rts_tmp)
+{
+  for (const auto & peptide : peptides)
+  {
+    if (!peptide.getHits().empty())
+    {
+      const String& sequence = peptide.getHits()[0].getSequence().toString();
+      double rt = peptide.getRT();
+      peptide_rts[sequence].push_back(rt);
+      rts_tmp.push_back(rt);
+    }
+  }
+}
+
+void TOPPMapAlignerTreeGuided::extractSeqAndRt_(const vector<FeatureMap>& feature_maps,
+        vector<SeqAndRTList>& maps_seq_and_rt, vector<vector<double>>& maps_ranges)
+{
+  for (Size i = 0; i < feature_maps.size(); ++i)
+  {
+    vector<double> rts_tmp(feature_maps[i].size());
+    for (auto feature_it = feature_maps[i].begin(); feature_maps[i].end() != feature_it; ++feature_it)
+    {
+      if (!feature_it->getPeptideIdentifications().empty())
+      {
+        getPeptideSequences_(feature_it->getPeptideIdentifications(), maps_seq_and_rt[i], rts_tmp);
+      }
+    }
+    sort(rts_tmp.begin(), rts_tmp.end());
+
+    maps_ranges[i] = rts_tmp;
+    rts_tmp.clear();
```

Comment:
no need to clear. Only if `rts_tmp` is declared outside the loop (which might increase performance)